### PR TITLE
GitHub actions

### LINKED ISSUE #16: Migration to GitHub actions
Right now the release process is initiated manually or via my personal Jenkins.

I want to set up GitHub actions to do this in the future.

### PR DESCRIPTION
**Pull request by 616slayer616**
- Added GitHub actions workflows
- Removed Jenkinsfile
- Removed travis workflow

### CODE REVIEW

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        with:
+          fetch-depth: 0
+
+      - name: build
+        run: ./gradlew clean assemble
+
+      - name: test
+        run: ./gradlew check
+
+      - name: release
```
Comment (timon-schelling) [inline .github/workflows/master.yml]: Change name to prerelease or something?

File: .github/workflows/feature.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        with:
+          fetch-depth: 0
+
+      - name: build
+        run: ./gradlew clean assemble
+
+      - name: test
+        run: ./gradlew check
+
+      - name: release
```
Comment (timon-schelling) [inline .github/workflows/feature.yml]: Release in feature?

File: .github/workflows/develop.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        with:
+          fetch-depth: 0
+
+      - name: build
+        run: ./gradlew clean assemble
+
+      - name: test
+        run: ./gradlew check
+
+      - name: release
```
Comment (timon-schelling) [inline .github/workflows/develop.yml]: Release in develop?

File: .github/workflows/develop.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  push:
+    branches:
+      - 'develop'
+
+jobs:
+  build-feature:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
```
Comment (timon-schelling) [inline .github/workflows/develop.yml]: Can name be set to "checkout"?

File: .github/workflows/feature.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    branches-ignore:
+      - 'develop'
+      - 'master'
+
+jobs:
+  build-feature:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
```
Comment (timon-schelling) [inline .github/workflows/feature.yml]: Can name be set to "checkout"?

File: .github/workflows/master.yml [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  push:
+    branches:
+      - 'master'
+      -
+jobs:
+  build-feature:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
```
Comment (timon-schelling) [inline .github/workflows/master.yml]: Can name be set to "checkout"?

File: .github/workflows/feature.yml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+        with:
+          fetch-depth: 0
+
+      - name: build
+        run: ./gradlew clean assemble
```
Comment (timon-schelling) [inline .github/workflows/feature.yml:21]: Isn't this  executed in an empty runner (no clean needed)?

File: .github/workflows/develop.yml [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+        with:
+          fetch-depth: 0
+
+      - name: build
+        run: ./gradlew clean assemble
```
Comment (timon-schelling) [inline .github/workflows/develop.yml:20]: Isn't this  executed in an empty runner (no clean needed)?

File: .github/workflows/master.yml [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+        with:
+          fetch-depth: 0
+
+      - name: build
+        run: ./gradlew clean assemble
```
Comment (timon-schelling) [inline .github/workflows/master.yml:20]: Isn't this  executed in an empty runner (no clean needed)?

**Comment by 616slayer616** [comment]
I'm not sure when or if the plugin will be transferred.
I would like to make the change to GitHub Actions asap.
We can change the deployment target later via the secrets.

File: .github/workflows/master.yml (hunk 1 above)
Comment (616slayer616) [reply .github/workflows/master.yml]: Forgot to change the name. This should be "sonar analysis"

File: .github/workflows/feature.yml (hunk 2 above)
Comment (616slayer616) [reply .github/workflows/feature.yml]: Forgot to change the name. This should be "sonar analysis"

File: .github/workflows/develop.yml (hunk 3 above)
Comment (616slayer616) [reply .github/workflows/develop.yml]: Forgot to change the name. This should be "sonar analysis"

File: .github/workflows/master.yml (hunk 9 above)
Comment (616slayer616) [reply .github/workflows/master.yml:20]: Yes. I wanted it to be identical to the previous Jenkins pipeline.
Some say always running clean is a best practice.

It doesn’t hurt in any case

File: .github/workflows/develop.yml (hunk 8 above)
Comment (616slayer616) [reply .github/workflows/develop.yml:20]: Yes. I wanted it to be identical to the previous Jenkins pipeline.
Some say always running clean is a best practice.

It doesn’t hurt in any case

File: .github/workflows/feature.yml (hunk 7 above)
Comment (616slayer616) [reply .github/workflows/feature.yml:21]: Yes. I wanted it to be identical to the previous Jenkins pipeline.
Some say always running clean is a best practice.

It doesn’t hurt in any case

File: .github/workflows/develop.yml (hunk 4 above)
Comment (616slayer616) [reply .github/workflows/develop.yml]: no there is checkout@v1 but that's an old version

File: .github/workflows/feature.yml (hunk 5 above)
Comment (616slayer616) [reply .github/workflows/feature.yml]: no there is checkout@v1 but that's an old version

File: .github/workflows/master.yml (hunk 6 above)
Comment (616slayer616) [reply .github/workflows/master.yml]: no there is checkout@v1 but that's an old version

**Comment by 616slayer616** [comment]
Ownership has been transferred.
The secrets should now work as intended

**Review by timon-schelling — APPROVED** [review]

**Pull request merged**